### PR TITLE
[#59] [Part 2] As a developer, I can have a CD process to build and deploy the iOS app to TestFlight

### PR DESCRIPTION
**Pull request by manh-t**
https://github.com/nimblehq/flutter_templates/issues/59

## What happened 👀

As a developer, we need to prepare a CD process to build and deploy the staging iOS app to TestFlight.

## Insight 📝

Write WIKI for `iOS Deployment Process`

## Proof Of Work 📹

n/a

### CODE REVIEW

File: .github/wiki/Deployment.md [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
\ No newline at end of file
+## For iOS
+
+### Guide
+
+- Before deploying the iOS application, we have to prepare the necessary credentials and information:
+
+    - An Apple developer account.
+    - Your application’s bundle id. This could be multiple bundle ids according to the number of application’s flavors.
+    - An application that links to the bundle id.
```
Comment (luongvo) [inline .github/wiki/Deployment.md:11]: What does this mean? Do you mean it's our project build for app store publishing? Is it a redundant thing to note here? 💭

Comment (luongvo) [reply .github/wiki/Deployment.md:11]: Or do you mean we need to setup a new app on App Store Connect? This sentence is quite vague 💭

File: .github/wiki/Deployment.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+    - The access to Github’s [Match repository](https://codesigning.guide/).
```
Comment (luongvo) [inline .github/wiki/Deployment.md:12]: it could be any git repo, not always Github repo here 🙏

File: .github/wiki/Deployment.md [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+    - **FASTLANE_SESSION**: When your Apple ID has Two-Factor Authentication or Two-Step verification information, we can’t validate your Apple ID with a prompt on your CI machine. We need to generate a login session for Apple ID in advance with `spaceauth`. Please check `Method 2: Two-step or two-factor authentication` in [here](https://docs.fastlane.tools/getting-started/ios/authentication/) to get the fastlane session and read the note in `Important note about session duration`.
+    - **MATCH_PASSWORD**: This will be referred to internally in `fastlane match` to decrypt your profiles. We can get the password from the one who has Admin access in Apple Store Connect.
+    - **KEYCHAIN_PASSWORD**: We can create a keychain password for ourself and it could be anything.
+    - **SSH_PRIVATE_KEY**: In order to fetch the distribution files from Github’s [Match repository](https://codesigning.guide/), we have to [generate an SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) and save it as the environment variable in CI.
```
Comment (luongvo) [inline .github/wiki/Deployment.md:24]: same, not always Github

File: .github/wiki/Deployment.md (hunk 1 above)
Comment (manh-t) [reply .github/wiki/Deployment.md:11]: > Or do you mean we need to setup a new app on App Store Connect? This sentence is quite vague 💭

@luongvo yeah, I mean that it's a new app on App Store Connect 😄 Maybe I will rewrite the sentence to `A new app's on App Store Connect that links to the bundle id.` What do you think?

Comment (luongvo) [reply .github/wiki/Deployment.md:11]: @manh-t Sounds better

**Comment by manh-t** [comment]
@luongvo updated 🙏

**Review by luongvo — APPROVED** [review]
lgtm

**Comment by manh-t** [comment]
@markgravity @doannimble can I have your reviews and approvals for this PR? 🐤

**Review by doannimble — APPROVED** [review]
LGTM 💯

**Pull request merged**